module/VuFind/src/VuFind/Config/Upgrade.php

```diff
@@ -232,13 +255,17 @@ protected function loadConfigs(): void
      * Apply settings from an old configuration to a new configuration.
      *
      * @param string $filename     Name of the configuration being updated.
-     * @param array  $fullSections Array of section names that need to be fully
+     * @param ?array $fullSections Array of section names that need to be fully
      * overridden (as opposed to overridden on a setting-by-setting basis).
      *
      * @return void
      */
-    protected function applyOldSettings(string $filename, array $fullSections = []): void
+    protected function applyOldSettings(string $filename, ?array $fullSections = null): void
     {
+        if ($fullSections === null) {
+            $fullSections = $this->defaultFullSections;
+        }
```

Comment:
I believe you could simplify to:

```suggestion
        $fullSections ??= $this->defaultFullSections;
```